rotkehlchen/chain/evm/decoding/pendle/utils.py

```diff
@@ -0,0 +1,114 @@
+                    log.warning(f'Skipping pendle market entry {market} due to {msg}')
+                    continue
+
+    except (requests.RequestException, TypeError, KeyError) as e:
+        msg = f'missing key {e!s}' if isinstance(e, KeyError) else f'{e!s}'
+        log.error(f'Failed to retrieve Pendle markets for chain {chain} due to {msg}')
+
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.PENDLE_POOLS, str(chain.serialize())),
+            values=all_markets,
+        )
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.PENDLE_SY_TOKENS, str(chain.serialize())),
+            values=all_sy_tokens,
+        )
+
+
+def query_pendle_price(token: 'EvmToken') -> Price:
+    """Query USD price for a given Pendle PT/YT/SY/LP token"""
+    try:
+        result = request_get_dict(f'https://api-v2.pendle.finance/core/v1/{token.chain_id.serialize()}/assets/prices?addresses={token.evm_address}')
+        return deserialize_price(result['prices'][token.evm_address.lower()])
+    except (RemoteError, UnableToDecryptRemoteData, DeserializationError, TypeError, ValueError, KeyError) as e:  # noqa: E501
+        msg = f'missing key {e!s}' if isinstance(e, KeyError) else f'{e!s}'
+        log.error(f'Unable to query usd price of pendle wrapped token {token} due to {msg}')
+        return ZERO_PRICE
+
+
+def query_pendle_yield_tokens(evm_inquirer: 'EvmNodeInquirer') -> None:
+    """Queries all Pendle yield tokens(LP, SY, PT, YT) and saves them to allow price queries."""
+    all_tokens: set[ChecksumEvmAddress] = set()
+    try:
+        all_tokens.update(deserialize_evm_address(entry['address']) for entry in request_get_dict(f'https://api-v2.pendle.finance/core/v3/{evm_inquirer.chain_id.serialize()}/assets/all'))
+    except (RemoteError, UnableToDecryptRemoteData, DeserializationError, IndexError, KeyError) as e:  # noqa: E501
+        msg = f'missing key {e!s}' if isinstance(e, KeyError) else f'{e!r}'
+        log.error(f'Unable to fetch pendle data for {evm_inquirer.chain_name} due to {msg}')
+        return
+
+    for token in all_tokens:
+        try:
+            cached_token = get_token(evm_address=token, chain_id=evm_inquirer.chain_id)
+            get_or_create_evm_token(
+                evm_inquirer=evm_inquirer if cached_token is None else None,
+                name=cached_token.name if cached_token is not None else None,
+                symbol=cached_token.symbol if cached_token is not None else None,
+                decimals=cached_token.decimals if cached_token is not None else None,
+                protocol=CPT_PENDLE,
+                userdb=evm_inquirer.database,
+                evm_address=token,
+                chain_id=evm_inquirer.chain_id,
+            )
```

Comment:
You are repeating the same if check 4 times per iteration in this for loop. 

```suggestion
            if (cached_token := get_token(evm_address=token, chain_id=evm_inquirer.chain_id)) is not None:
                    get_or_create_evm_token(
                        evm_inquirer=evm_inquirer,
                        name=cached_token.name,
                        symbol=cached_token.symbol,
                        decimals=cached_token.decimals,
                        protocol=CPT_PENDLE,
                        userdb=evm_inquirer.database,
                        evm_address=token,
                        chain_id=evm_inquirer.chain_id,
                   )
              else:
                    get_or_create_evm_token(
                        evm_inquirer=None
                        name=None
                        symbol=None,
                        decimals=None,
                        protocol=CPT_PENDLE,
                        userdb=evm_inquirer.database,
                        evm_address=token,
                        chain_id=evm_inquirer.chain_id,
                   )
```